### PR TITLE
Pass prop isActiveTab onto each child view

### PR DESCRIPTION
**Pull request by mosesoak**
Gives child views an easy way to update themselves each time they
are shown. This is possible to do without this modification, if a state
is kept in the parent of the tabview and wired to each child manually,
the point of this PR is to automate this and reduce that overhead.

### CODE REVIEW

**Comment by skv-headless** [comment]
This is definitely useful but it would be difficult to find for newcomers. Also I like python zen https://www.python.org/dev/peps/pep-0020/ .
quote from there
*There should be one-- and preferably only one --obvious way to do it*

File: index.js [hunk 1]
```diff
@@ -113,7 +113,7 @@ const ScrollableTabView = React.createClass({
             return <View
               key={child.props.tabLabel + '_' + idx}
               style={{width: this.state.containerWidth, }}>
-              {child}
+              { React.cloneElement(child, { isActiveTab:(idx === this.state.currentPage)}) }
```
Comment (mosesoak) [inline index.js]: React.cloneElement is already used in this way in this same component, see:
https://github.com/brentvatne/react-native-scrollable-tab-view/blob/master/index.js#L76

In my app, no other existing props were disrupted when adding this one. It was actually my heavy use of this component that instilled my confidence in using cloneElement as a solid technique.

**Comment by mosesoak** [comment]
also, thanks for your time considering it. :smile:

**Comment by skv-headless** [comment]
I need time to think

**Comment by jdzak** [comment]
> This is definitely useful but it would be difficult to find for newcomers. Also I like python zen https://www.python.org/dev/peps/pep-0020/ .
quote from there
There should be one-- and preferably only one --obvious way to do it

@skv-headless Is there already a way to know when a tab is selected for a page?

I'm currently trying to figure out how to do this because I have a an app with multiple pages that send API requests which are all fired when the `ScrollTabView` is rendered. When the user navigates to another page, I'd like to refresh that page (send another API request), so they don't see the stale data from the initial render.

**Comment by mosesoak** [comment]
@jdzak The parent of the tab view can listen for changes, retain its own state, and pass that as a prop to each child tab. This PR would just make that easier since an app with many such views is simplified by providing that information to the child views out of the box. Glad to know we're not alone! :-)

**Comment by jdzak** [comment]
> @jdzak The parent of the tab view can listen for changes, retain its own state, and pass that as a prop to each child tab. This PR would just make that easier since an app with many such views is simplified by providing that information to the child views out of the box. Glad to know we're not alone! :-)

@mosesoak Would the best way to currently do this be to subclass `ScrollableTabView`?

**Comment by skv-headless** [comment]
My way 
```
const SimpleExample = React.createClass({
  render() {
    return <ScrollableTabView onChangeTab={({i, ref}) => this.setState({currentTab: i})}>
      <Text tabLabel='Tab #1' isCurrent={this.state.currentTab === 0}>My</Text>
      <Text tabLabel='Tab #2'>favorite</Text>
      <Text tabLabel='Tab #3'>project</Text>
    </ScrollableTabView>;
  },
});
```

**Comment by BlakeWilliams** [comment]
My app would also be simplified by this. Doing it manually is fine too, but having it passed down would be a nice addition. I also think that as long as there's good documentation beginners will be fine. I don't think they would notice that there's an extra prop passed down.

**Comment by mosesoak** [comment]
Not sure if there's a product pain entry for viewDid/WillAppear hooks but it certainly would be a great addition to the core framework, something I miss very much from iOS. @skv-headless I like your solution, it's nice and clean.

**Comment by mosesoak** [comment]
(Except that it requires adding states in many containers around our app, so I'll continue to use my branch on this.)

**Comment by skv-headless** [comment]
BTW what for are you using it? maybe we can find better solution?

**Comment by mosesoak** [comment]
@skv-headless We have a substantial app that we wanted to add animation triggers on each page view. For a page that contains a scrollable-tab-view, the content is within the child view. I figured out ways to make each child view be triggered on-view in two cases, a) when navigating away from the top-level page and then back, I pass a prop down indicating the parent focus state and b) when you change between tabs.

So I actually needed two props to be passed down to make my system fully automated. Maybe we could spread a generic props object to each child view, like childViewProps={{...}} instead to allow for expansion...?

**Comment by kuma-guy** [comment]
@skv-headless 

Hi, I have the use case similar to @jdzak and want ```isCurrent``` prop for each tab,
but it seems to that each tab does not receive ``` isCurrent``` prop when ```componentWillReceiveProps```

my code are below, what am I missing?

https://gist.github.com/kuma-guy/bd2bea1e83dea990679cd14fa6890821

**Comment by ptomasroos** [comment]
This PR is very outdated, hence if this changes are still needed feel free to rebase them on top of master and open a new PR

**Pull request closed without merging**